docs/upgrade.html

```diff
@@ -29,7 +31,26 @@ <h5><a id="upgrade_410_notable" href="#upgrade_410_notable">Notable changes in 4
                 </ul>
             </li>
         </ul>
-<h4><a id="upgrade_4_0_0" href="#upgrade_4_0_0">Upgrading to 4.0.0 from any version 3.3.x through 3.9.x</a></h4>
+
+<h4><a id="upgrade_4_0_0" href="#upgrade_4_0_0">Upgrading to 4.0.0</a></h4>
+
+<h5><a id="upgrade_clients_4_0_0" href="#upgrade_clients_4_0_0">Upgrading Clients to 4.0.0</a></h5>
+
+<p><b>For a rolling upgrade:</b></p>
+
+<ol>
+    <li>Upgrade the clients one at a time: shut down the client, update the code, and restart it.</li>
+    <li>For the Kafka client upgrade path, note that many deprecated APIs were removed in Kafka 4.0. Additionally, upgrading directly to 4.x from certain versions is not feasible.
+        For more information, please refer to <a href="https://cwiki.apache.org/confluence/x/y4kgF">KIP-1124</a>.</li>
```

Comment:
I wonder whether we should call out that 2.1 is the minimum required version here KIP-896.

Reply:
Also, "For the Kafka client upgrade path" is redundant in point 2. This whole section is about client upgrades, right?

Reply:
I would use the following:

> 2. Clients (including Streams and Connect) must be on version 2.1 or higher before upgrading to 4.0. Many deprecated APIs were removed in Kafka 4.0. For more information about the compatibility, please refer to compatibility matrix (link it) or KIP-1124 for details.

Reply:
@mingdaoy I hope you don't mind but I push a small commit to tweak the text.

Reply:
@dajac you were one second faster than me. Thank you for the fix.